src/main/java/com/synopsys/integration/alert/channel/email/descriptor/EmailDistributionDescriptorActionApi.java

```diff
@@ -72,56 +73,66 @@ public void validateChannelConfig(final FieldAccessor fieldAccessor, final Map<S
-                                           .collect(Collectors.toSet());
+        if (providerName.isPresent() && BlackDuckProvider.COMPONENT_NAME.equals(providerName.get())) {
+            Set<BlackDuckProjectEntity> blackDuckProjectEntities = null;
+            if (filterByProject.isPresent() && BooleanUtils.toBoolean(filterByProject.get())) {
+                final Set<String> configuredProjects = fieldModel.getField(BlackDuckDescriptor.KEY_CONFIGURED_PROJECT).getValues().stream().collect(Collectors.toSet());
+                final Optional<String> projectNamePattern = fieldModel.getField(BlackDuckDescriptor.KEY_PROJECT_NAME_PATTERN).getValue();
+                blackDuckProjectEntities = blackDuckProjectRepositoryAccessor.readEntities()
+                                               .stream()
+                                               .map(databaseEntity -> (BlackDuckProjectEntity) databaseEntity)
+                                               .filter(databaseEntity -> doesProjectNameMatchThePattern(databaseEntity.getName(), projectNamePattern.orElse(""))
+                                                                             || doesProjectNameMatchAConfiguredProject(databaseEntity.getName(), configuredProjects))
+                                               .collect(Collectors.toSet());
+            } else {
+                blackDuckProjectEntities = blackDuckProjectRepositoryAccessor.readEntities()
+                                               .stream()
+                                               .map(databaseEntity -> (BlackDuckProjectEntity) databaseEntity)
+                                               .collect(Collectors.toSet());
 
-        }
-        if (null != blackDuckProjectEntities) {
-            final Optional<String> projectOwnerOnlyOptional = fieldModel.getField(EmailDescriptor.KEY_PROJECT_OWNER_ONLY).getValue();
-            final Boolean projectOwnerOnly = Boolean.parseBoolean(projectOwnerOnlyOptional.orElse("false"));
-            final Set<String> projectsWithoutEmails = new HashSet<>();
-            blackDuckProjectEntities
-                .stream()
-                .forEach(project -> {
-                    final Set<String> emailsForProject = emailGroupChannel.getEmailAddressesForProject(project, projectOwnerOnly);
-                    if (emailsForProject.isEmpty()) {
-                        projectsWithoutEmails.add(project.getName());
+            }
+            if (null != blackDuckProjectEntities) {
+                final Optional<String> projectOwnerOnlyOptional = fieldModel.getField(EmailDescriptor.KEY_PROJECT_OWNER_ONLY).getValue();
+                final Boolean projectOwnerOnly = Boolean.parseBoolean(projectOwnerOnlyOptional.orElse("false"));
+                final Set<String> projectsWithoutEmails = new HashSet<>();
+                blackDuckProjectEntities
+                    .stream()
+                    .forEach(project -> {
```

Comment:
I think we want to avoid using `Collection.forEach` for non-trivial operations. As a side note, I don't think `.stream().forEach()` buys us anything over a `.forEach()` without a stream unless we are going from an "ordered" collection and the inner logic doesn't care about order.